### PR TITLE
feat(cli): read and write ipfs url for decentralization

### PR DESCRIPTION
**Pull request by dbeal-eth**
note: publishipfsurl/cannon repo is still separate. and optional if not publishing.

### CODE REVIEW

File: packages/cli/src/settings.ts [hunk 1]
```diff
@@ -37,9 +37,14 @@ export type CliSettings = {
   ipfsTimeout?: number;
 
   /**
-   * the url of the IPFS endpoint to use as a storage base. defaults to localhost IPFS
+   * the url of the IPFS endpoint to use as a storage base, only for reading data
    */
-  ipfsUrl?: string;
```
Comment (noahlitvin) [inline packages/cli/src/settings.ts:42]: I thought the idea is we leave this and deprecate it? I think it'd be a fallback for readIpfsUrl?

**Comment by noahlitvin** [comment]
Question on fallback behavior:

I think for read, we'd want to try filesystem/cache, `readIpfsUrl` (if set), also `ipfsUrl` (deprecated, but if present), and also default repo url?

For write, it goes to local cache, and writes to `writeIpfsUrl` if present (*not* the deprecated `ipfsUrl`)

For publish, it goes to `publishIpfsUrl` if present, otherwise the default repo url. Should it also write to the repo url when the `publishIpfsUrl` is set to something?

File: packages/cli/src/settings.ts (hunk 1 above)
Comment (dbeal-eth) [reply packages/cli/src/settings.ts:42]: really like to avoid keeping it around if we are taking this direction

**Comment by dbeal-eth** [comment]
> ant to try files

everything that you say here should be what its doign with this change, minus the existance of `ipfsUrl` setting

**Review by FuzzB0t — APPROVED** [review]

**Pull request merged**